pkg/azure/api/api.go

```diff
@@ -63,27 +63,19 @@ type MetricsAPI interface {
 	ObserveRateLimit(operation string, duration time.Duration)
 }
 
-func constructAuthorizer(cloudName, userAssignedIdentityID string) (autorest.Authorizer, error) {
+func constructAuthorizer(env azure.Environment, userAssignedIdentityID string) (autorest.Authorizer, error) {
 	if userAssignedIdentityID != "" {
-		env, err := azure.EnvironmentFromName(cloudName)
-		if err != nil {
-			return nil, err
-		}
-		msiEndpoint, err := adal.GetMSIVMEndpoint()
+		spToken, err := adal.NewServicePrincipalTokenFromManagedIdentity(env.ServiceManagementEndpoint, &adal.ManagedIdentityOptions{
+			IdentityResourceID: userAssignedIdentityID,
```

Comment:
> Move to using NewServicePrincipalTokenFromManagedIdentity because the old method was deprecated.

I think this also changes the type of the required userAssignedIdentityID parameter:

Formerly, it was a GUID/UUID/ClientID. Here, a full Azure resource id is expected instead. I think this should retain the former behaviour:

```suggestion
			ClientID: userAssignedIdentityID,
```

Reply:
Good catch, I updated the code.